src/main/java/org/broadinstitute/dsm/cf/TestBostonKitTrackerDispatcher.java

```diff
@@ -0,0 +1,145 @@
+import com.typesafe.config.Config;
+import lombok.NonNull;
+import org.apache.commons.dbcp2.PoolableConnection;
+import org.apache.commons.dbcp2.PoolingDataSource;
+import org.broadinstitute.dsm.db.DDPInstance;
+import org.broadinstitute.dsm.db.DdpKit;
+import org.broadinstitute.dsm.model.ups.UPSStatus;
+import org.broadinstitute.dsm.pubsub.KitTrackerPubSubPublisher;
+import org.broadinstitute.dsm.statics.DBConstants;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Identifies kits whose shipping history should be
+ * queried (by another job) in order to produce
+ * reports on turnaround time and study adherence.
+ */
+public class TestBostonKitTrackerDispatcher implements BackgroundFunction<PubsubMessage> {
+
+    private static final Logger logger = Logger.getLogger(TestBostonKitTrackerDispatcher.class.getName());
+
+    private String STUDY_MANAGER_SCHEMA = System.getenv("STUDY_MANAGER_SCHEMA") + ".";
+    private int LOOKUP_CHUNK_SIZE;
+
+
+    KitTrackerPubSubPublisher kitTrackerPubSubPublisher = new KitTrackerPubSubPublisher();
+
+
+    @Override
+    public void accept(PubsubMessage pubsubMessage, Context context) throws Exception {
+        Config cfg = CFUtil.loadConfig();
+        String dbUrl = cfg.getString("dsmDBUrl");
+
+        PoolingDataSource<PoolableConnection> dataSource = CFUtil.createDataSource(20, dbUrl);
+        final String SQL_SELECT_KITS = "SELECT kit.dsm_kit_request_id,req.ddp_instance_id, kit.kit_label, kit.tracking_to_id, kit.tracking_return_id, kit.error, kit.message, kit.receive_date, " +
+                "kth.kit_shipping_history,  kth.kit_return_history,  req.bsp_collaborator_participant_id, kit.ups_tracking_status, kit.ups_return_status, kit.ups_tracking_date, kit.ups_return_date, " +
+                " req.external_order_number, kit.CE_order FROM " + STUDY_MANAGER_SCHEMA + "ddp_kit kit LEFT JOIN " + STUDY_MANAGER_SCHEMA + "ddp_kit_request req " +
+                " ON (kit.dsm_kit_request_id = req.dsm_kit_request_id) " +
+                "left join " + STUDY_MANAGER_SCHEMA + "kit_tracking_history kth on ( kth.dsm_kit_request_id = kit.dsm_kit_request_id) " +
+                " WHERE req.ddp_instance_id = ? and kit_label not like \"%\\\\_1\"  ";
+        String SQL_AVOID_DELIVERED = " and (tracking_to_id is not null or tracking_return_id is not null ) and (kth.kit_shipping_history is null or kth.kit_return_history is null or kth.kit_shipping_history not like \"" + UPSStatus.DELIVERED_TYPE + " %\" or kth.kit_return_history not like \"" + UPSStatus.DELIVERED_TYPE + " %\")" +
+                " order by kit.dsm_kit_request_id ASC";
+
+        logger.info("Starting the UPS lookup job");
+        LOOKUP_CHUNK_SIZE = cfg.getInt("kitTracking.trackingPerCFSize"); // todo pegah put in constants
```

Comment:
Rather than having the chunk size set in the config file--which requires deployment--let's read it from the pubsub message itself that is passed into `accept()`.  This way, you can fiddle with it in whatever starts the job without having to redeploying the job.